contracts/validator/ValidatorSet.sol

```diff
@@ -89,4 +98,69 @@ contract ValidatorSet {
             validatorEndHeight[_validator] > _height &&
             validatorEndHeight[_validator] > 0;
     }
+
+
+    /* Internal Functions  */
+
+    /**
+     * @notice It is for inserting validators into the validator set.
+     *
+     * @dev Function requires :
+     *          - Validator address must not be 0.
+     *          - Validator address is already not used.
+     *
+     * @param _validator Validator address.
+     * @param _beginHeight Begin height for the validator.
+     */
+    function insertValidatorInternal(
+        address _validator,
+        uint256 _beginHeight
+    )
+        internal
+    {
+        require(
+            _validator != address(0),
+            "Validator must not be null address."
+        );
+        require(
+            validatorBeginHeight[_validator] == 0 && validatorEndHeight[_validator] == 0,
+            "Validator address is already used."
+        );
+
+        address currentValidator = validators[SENTINEL_VALIDATORS];
```

Comment:
`currentValidator` is a bad name. You don't have to write this to memory, which is the best solution. If you do want to name it, name it eg. `lastValidator` because Sentinel points to the last element in the linked list.